src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs

```diff
@@ -131,6 +131,36 @@ public virtual StringBuffer Render(IMarkdownRenderer renderer, DfmVideoBlockSpli
             StringBuffer content = string.Empty;
 
             var videoToken = splitToken.Token as DfmVideoBlockToken;
+            //TODO: fix video link
+            //if youtube.com, change to youtube-nocookie.com
+            //if channel9.msdn.com, add ?nocookie=true at end
+            if (videoToken != null && !String.IsNullOrWhiteSpace(videoToken.Link))
+            {
+                string link = videoToken.Link;
+                Uri videoLink;
+                if (Uri.TryCreate(link, UriKind.Absolute, out videoLink))
+                {
+                    string host = videoLink.Host;
+
+                    if (host.Equals("channel9.msdn.com", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        if (link.Contains("?"))
+                        {
+                            link = link + "&nocookie=true";
+                        }
+                        else
+                        {
+                            link = link + "?nocookie=true";
+                        }
+                    }
+                    else if (host.IndexOf("youtube.com", StringComparison.InvariantCultureIgnoreCase) > -1)
+                    {
+                        link = link.Replace("youtube.com", "youtube-nocookie.com");
```

Comment:
>link.Replace [](start = 31, length = 12)

use Regex.Replace for ignore case?

Reply:
Instantiating a Regex object seems heavier than IndexOf performance wise, should I run a perf test between the two paths?

Reply:
Updated the code based on the other pieces of feedback (line break between methods, string instead of String)

Reply:
Yes, performance should be an issue. My concern is that the ignore case `IndexOf` check makes no sense if the `Replace` is case sensitive...